cmd/image-builder-db-test/main_test.go

```diff
@@ -376,6 +376,42 @@ func testBlueprints(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, b1, b2)
 
+	updated := v1.BlueprintV1{
+		Version:     2,
+		Name:        "new name",
+		Description: "new desc",
+	}
+	bodyUpdated1, err := json.Marshal(updated)
+	require.NoError(t, err)
+
+	newVersionId := uuid.New()
+	err = d.UpdateBlueprint(newVersionId, entry.Id, ORGID1, "new name", "new desc", bodyUpdated1)
+	require.NoError(t, err)
+	entryUpdated, err := d.GetBlueprint(entry.Id, ORGID1, ANR1)
+	require.NoError(t, err)
+	bodyUpdated2, err := v1.BlueprintFromEntry(entryUpdated)
+	require.NoError(t, err)
+	require.Equal(t, updated, bodyUpdated2)
+	require.NotEqual(t, b1, bodyUpdated2)
+
+	newBlueprintVersion := v1.BlueprintV1{
+		Version:     3,
+		Name:        "name should not be changed",
+		Description: "desc should not be changed",
+	}
+	newBlueprintVersionBody, err := json.Marshal(newBlueprintVersion)
+	require.NoError(t, err)
+	newBlueprintId := uuid.New()
+	err = d.UpdateBlueprint(newBlueprintId, entry.Id, ORGID2, "name", "desc", newBlueprintVersionBody)
+	require.Error(t, err)
+	entryAfterInvalidUpdate, err := d.GetBlueprint(entry.Id, ORGID1, ANR1)
+	require.NoError(t, err)
+	bodyNotUpdated, err := v1.BlueprintFromEntry(entryAfterInvalidUpdate)
+	require.NoError(t, err)
+	require.Equal(t, updated, bodyNotUpdated)
+
+	require.NotEqual(t, b1, bodyNotUpdated)
+
```

Comment:
Also test `version` column (not only the version stored in the JSON).

I think this test reveals that storing blueprint version in the column AND in the body column is counter-productive. I suggest to remove it from the `BlueprintV1` type completely, we need to copy the data anyway into OpenAPI specification type so we can copy from the column rather than from the body content.

Reply:
Should I do it in this PR, or do we do this in a followup?

Reply:
I'd do that in a follow-up :)

Reply:
I'd say that description and name are the same case, we do have columns for those as well :)

Reply:
Good call, adding it as an item here: https://issues.redhat.com/browse/HMS-3308